relayer/src/chain/handle/prod.rs

```diff
@@ -0,0 +1,275 @@
+use std::fmt::Debug;
+
+use crossbeam_channel as channel;
+
+use ibc::{
+    ics02_client::client_def::{AnyClientState, AnyConsensusState, AnyHeader},
+    ics03_connection::connection::ConnectionEnd,
+    ics04_channel::channel::ChannelEnd,
+    ics23_commitment::commitment::CommitmentPrefix,
+    ics23_commitment::merkle::MerkleProof,
+    ics24_host::identifier::ChainId,
+    ics24_host::identifier::ChannelId,
+    ics24_host::identifier::{ClientId, ConnectionId, PortId},
+    proofs::Proofs,
+    Height,
+};
+
+// FIXME: the handle should not depend on tendermint-specific types
+use tendermint::account::Id as AccountId;
+
+use crate::{
+    chain::QueryResponse,
+    error::{Error, Kind},
+    keyring::store::KeyEntry,
+    tx::connection::ConnectionMsgType,
+};
+
+use super::{reply_channel, ChainHandle, HandleInput, ReplyTo, Subscription};
+
+#[derive(Debug, Clone)]
+pub struct ProdChainHandle {
+    chain_id: ChainId,
+    sender: channel::Sender<HandleInput>,
```

Comment:
Two remarks:

1. As we discussed, it might be a good idea to find a different name for `HandleInput`. If I understand correctly, a `HandleInput` type is what the `ProdChainHandle` sends to the `ChainRuntime`. So this type is an input to the runtime, which the handle produces (alternative names would be: `RuntimeInput` or HandleRequest`).

2. Took me some time to understand what the `sender` does. We used the same name for the local var `sender` (in the method `send` below) which added to my confusion. It's clear that this is a sender by the type `channel::Sender..` here, so perhaps we can give this field a more accurate name, like `channel_to_runtime` or `runtime_sender` or just `runtime` or something like that. This is a more general comment that [applies to the runtime](https://github.com/informalsystems/ibc-rs/blob/052d4f7a906feaac89d6c9e3eee6aa5dc436e520/relayer/src/chain/runtime.rs#L54) as well, where we have the fields `sender` and `receiver` (the field `event_receiver` was very clear, in contrast).

Reply:
1. Fully agree. How about `ChainRequest`?
2. Totally! I like `runtime_sender` or `runtime_tx` (`tx` seems to be a common choice for a `Sender`, although I agree that it's not as clear as `sender`), or perhaps just `to_runtime`?

Reply:
> * Fully agree. How about `ChainRequest`?

perfect.

> * Totally! I like `runtime_sender` or `runtime_tx` (`tx` seems to be a common choice for a `Sender`, although I agree that it's not as clear as `sender`), or perhaps just `to_runtime`?

`runtime_sender` seems to be the most aligned with other places (e.g., `event_receiver`); `to_runtime` is a second best option in my opinion!